File: codewars/meeting.fs
open System

let toName(s: string) =
  let x = s.Split ':'
  x.[1] + ", " + x.[0]
  
let meeting(s: string): string =
  s.Split ';'
    |> Array.map (toName >> (fun x -> x.ToUpper()))
    |> Array.sort
    |> Array.fold (fun acc item -> acc + "(" + item + ")") ""

File: codewars/src/Meeting.hs
module Meeting (meeting) where

import Data.Char
import Data.Function
import Data.List
import Data.List.Split

meeting :: String -> String
meeting s =
  s
    & map toUpper
    & splitOn ";"
    & map toName
    & sortBy (flip compare)
    & foldr (\x acc -> acc ++ "(" ++ x ++ ")") ""

toName :: String -> String
toName s =
  let (firstname : lastname : []) = splitOn ":" s
   in lastname ++ ", " ++ firstname
